themes/basic_eww/eww-bar/eww.xml

```diff
@@ -0,0 +1,101 @@
+        <box orientation="h" class="workspaces" space-evenly="true" halign="start"> <!-- Your workspaces.-->
+            <!-- Attention a literal only takes one element as content, so if you need more elements put them in a <box> -->
+          <literal content="{{leftwm-tags}}"/>
+        </box>
+    </def>
+    <def name="music">
+        <box orientation="h" class="music" halign="center" space-evenly="false">
+          <!-- An image. We won't be using it, but it might be handy if you ever need it. -->
+          <!-- <image path="/home/user/.config/eww/icons/music.svg"/> -->
+
+             {{music}} <!--This is how you use variables.-->
+        </box>
+    </def>
+    <def name="slider-vol">
+      <!--A basic volume slider-->
+      <box orientation="h" class="slider-vol" space-evenly="false">
+        <box class="label-vol"></box>
+        <scale min="0" max="101" value="{{volume}}" onchange="amixer -D pulse sset Master {}%"/>
+      </box>
+    </def>
+    <def name="slider-ram">
+      <!--A basic RAM slider-->
+      <box orientation="h" class="slider-ram" space-evenly="false">
+        <box class="label-ram"></box>
+        <scale min="0" active="false" max="101" value="{{ram-used}}"/>
+      </box>
+    </def>
+    <def name="time">
+      <!--Time-->
+      <box class="time">{{hour}}:{{min}} {{month}} {{number_day}}, {{year_full}}</box>
+    </def>
+  </definitions>
+
+  <variables>
+
+	  <!-- Music vars -->
+    <!-- These are your variables. Like they say, they run a script, and then output that to the variable. In
+    the following var, we get music info from playerctl. -->
+
+    <script-var name="music" interval="5s">playerctl metadata --format '{{ artist }} - {{ title }}'</script-var>
+
+
+    <!--Volume var-->
+    <!-- If your vars get too complex to write down here, you can use an external script and paste it's path here as well, as shown below.
+     -->
+
+    <script-var name="volume" interval="16ms">~/.config/eww/scripts/getvol</script-var>
+
+    <!--Date vars-->
+    <script-var name="number_day" interval="5h">date "+%d"</script-var>
+    <script-var name="month" interval="10h">date "+%b"</script-var>
+    <script-var name="min" interval="10s">date "+%M"</script-var>
+    <script-var name="hour" interval="1m">date "+%H"</script-var>
+    <script-var name="year_full" interval="15h">date "+%Y"</script-var>
+
+    <!--RAM var-->
+    <script-var name="ram-used" interval="1s">~/.config/eww/scripts/getram</script-var>
+
+    <!--Battery charge var-->
+    <script-var name="battery-remaining" interval="2m">cat /sys/class/power_supply/BAT0/capacity</script-var>
+
+    <!-- Get the Tags of LeftWM -->
+    <!-- Note: a script-var without an interval takes the tail of the script command -->
+    <script-var name="leftwm-tags">leftwm-state -w 0 -t ~/.config/leftwm/themes/current/template.liquid</script-var>
+
+  </variables>
+
+  <windows>
+    <!-- These are the windows -->
+    <window name="bar" screen="0">
```

Comment:
Thanks!
This is basically the same way as on the `basic_polybar` right?

Initially I was hoping for something more elegant then 'hardcoding' the amount of bars but I believe that would need more logic to be possible inside `eww` configuration and I guess the addition of these logic capabilities won't be implemented before the intended config-language rework. 

So I probably should ad this as an example on how we could handle multiple workspaces for `eww` now and update for the elegant solution later.

Reply:
Exactly, I also tried to find some more elegant way, but it seems that this is the way peoples use (at least you can reuse widgets). It makes sense because eww is thought to do much more than only bars, but maybe there's a feature that I have overseen that handles this situation.

Reply:
Besides the use case of launching bars dynamically from within eww itself I was looking for more logic to have a button that cycles through some widgets, like 'layouts', 'windowtitle', 'music' etc, but so far I haven't found any..

Reply:
may I ask, how you work around the needed delay after starting `eww daemon`? Right now I am struggling to figure that out in combination with the `for` loop..

Reply:
for now I added a very uggly `sleep 1`, but it seems to work.